File: solutions/Hexspeak/solve/java/HexspeakSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class HexspeakSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        long ciphertext = (Long) reader.next(Long.class);
        reader.close();

        Hexspeak solver = new Hexspeak();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.decode(ciphertext));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}

File: solutions/Hexspeak/solve/cpp/HexspeakSolver.cpp
#include <fstream>
#include <topcoder>
#include "Hexspeak.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
    try {
        std::ifstream ifs(argv[1]);
        int64 ciphertext; tc::read(ifs, ciphertext);
        ifs.close();

        std::ofstream ofs(argv[2]);
        Hexspeak solver;
        tc::show(ofs, solver.decode(ciphertext));
        ofs.close();
    } catch (std::exception &e) {
        std::cerr << e.what() << std::endl;
    }
    return 0;
}